iceoryx_posh/source/popo/building_blocks/condition_listener.cpp

```diff
@@ -116,10 +109,8 @@ ConditionListener::NotificationVector_t ConditionListener::waitImpl(const cxx::f
 
 void ConditionListener::reset(const uint64_t index) noexcept
 {
-    if (index < MAX_NUMBER_OF_NOTIFIERS)
```

Comment:
Okay, seems to be safe since it is only called from the class itself. What do you think about calling such methods without bound checking`resetUnchecked`?